pkg/shell/active-pages-modal.jsx

```diff
@@ -57,14 +58,16 @@ export class ActivePagesDialog extends React.Component {
     }
 
     onRemove() {
+        const D = this.context;
         this.state.frames.forEach(element => {
             if (element.selected)
                 this.props.frames.remove(element.host, element.component);
         });
-        this.props.onClose();
+        D.close();
```

Comment:
Especially for such single-use calls, just `this.context.close()` may be enough? And if we want to point out the "data type",  i.e. that we expect the context to be a DialogContext, it should have a descriptive name.

Reply:
You mean `this.dialogs_context` instead of `this.context`?  I am afraid `this.context` is prescribed by React, or am I missing something?

Reply:
I meant that we do this here:
```js
const D = this.context;
D.close();
```
and not use `D` anywhere else. So obviously it *is* `this.context`, or what am I missing? I don't see anything magic here?

Reply:
Or do you *have* to assign `this.context` to a variable, and that's the magic that I'm missing?

Reply:
There is no magic.  I did `const D = this.context` only to give it a name, so that we are reminded what the context actually is.  This is all gone now with the rewrite to function components.